gobblin-core/src/main/java/gobblin/writer/MetadataAwareWriter.java

```diff
@@ -20,7 +20,12 @@
 
 
 /**
- * Represents a Writer that is metadata aware
+ * Represents a Writer that is metadata aware. Metadata aware writers
+ * specify default {@link GlobalMetadata} that will be added to all records that flow
+ * through the pipeline they are a part of.
+ *
+ * This allows a writer to communicate with the MetadataWriterWrapper that surrounds it
+ * and is the one responsible for processing any metadata that flows through the conversion pipeline.
  */
 public interface MetadataAwareWriter {
   GlobalMetadata getDefaultMetadata();
```

Comment:
Can you add a comment on the getDefaultMetadata method? 

What does default metadata mean?  Is it per dataset? 
Should the caller call this once and cache or call multiple times..